768: block glossary terms with word breaks

toolstack opened the pull request:
Resolves #768. 

Based on PR #745.

toolstack committed:
Block the creation of glossary entries that can never be matched.

toolstack committed:
Coding standards updates.

ocean90 reviewed (commented):
As mentioned on Slack, non-ascii characters can't be used for the original term anymore. But that's probably fine for now. Otherwise looks good.

toolstack commented:
I tracked down the non-ascii problem, it's a Unicode issue as when we call `preg_match()` we don't add the [`/u`](http://php.net/manual/en/reference.pcre.pattern.modifiers.php) parameter to the pattern which tells PHP to treat the pattern and subject as UTF8 strings.

toolstack commented:
This is the above mentioned Slack converstation: https://wordpress.slack.com/archives/C02RP4R9F/p1499190287524373

The reference to /u is not just in this PR, but in `map_glossary_entries_to_translations_originals()`, which does not match unicode terms either at the moment.

Testing a bit with `/u` causes more issues with word boundaries as it turns on several compatibility settings for Perl that seem to break word boundry detection :frowning_face:

As such, since matching UTF8 glossary terms does not work at the moment, I'll leave the `/u` option off and we can investigate the issue in another ticket.

I will update the error message to include verbiage around UTF-8 characters.

toolstack committed:
Update error message.

Multi-byte characters are also blocked by this patch as they are not matched via the existing glossary code.
